src/app/src/main/kotlin/org/imperial/mrc/hint/controllers/LoginController.kt

```diff
@@ -7,16 +7,25 @@ import org.springframework.ui.set
 import org.springframework.web.bind.annotation.GetMapping
 import javax.servlet.http.HttpServletRequest
 import org.imperial.mrc.hint.AppProperties
+import org.imperial.mrc.hint.security.oauth2.OAuth2AuthenticationRedirection
+import org.imperial.mrc.hint.security.oauth2.OAuth2State
 
 @Controller
-class LoginController(private val request: HttpServletRequest,
-                      private val session: Session,
-                      private val appProperties: AppProperties)
+class LoginController(
+    private val request: HttpServletRequest,
+    private val session: Session,
+    appProperties: AppProperties,
+    oauth2State: OAuth2State
+) : OAuth2AuthenticationRedirection(appProperties, oauth2State)
 {
-
     @GetMapping("/login")
-    fun login(model: Model): String
+    fun login(model: Model): Any
```

Comment:
Can we be more specific than `Any` here?

Reply:
> Auth0. Do we have an epic for adding e2e tests yet?

I am not sure who's working on this bit but I will check with Rob, I think it's going into next quarter milestone. Yes, it's tricky to test even something as simple as hitting auth0 endpoint.

Reply:
> Can we be more specific than `Any` here?

You are right but `Any` is a powerful alternative as I had like to return String and ResponseEntity.